modules/ballerina-core/src/main/java/org/ballerinalang/model/statements/AbstractStatement.java

```diff
@@ -58,4 +60,24 @@ public void setAlwaysReturns(boolean alwaysReturns) {
     public boolean isAlwaysReturns() {
         return alwaysReturns;
     }
+
+    @Override
+    public void setParent(Statement parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public Statement getParent() {
+        return this.parent;
+    }
+
+    @Override
+    public void setType(StatementType statementType) {
+        this.type = statementType;
+    }
+
+    @Override
+    public StatementType getType() {
```

Comment:
This getType() method will confuse developers with getType() usages in the other parts of the code.